lightwood/api/data_source.py

```diff
@@ -202,22 +199,35 @@ def __getitem__(self, idx):
 
         if not self.disable_cache:
             self.transformed_cache[idx] = sample
-            return self.transformed_cache[idx]
-        else:
-            return sample
+        return sample
 
     def get_column_original_data(self, column_name):
-        """
-
-        :param column_name:
-        :return:
-        """
         if column_name not in self.data_frame:
             nr_rows = self.data_frame.shape[0]
             return [None] * nr_rows
 
         return self.data_frame[column_name].tolist()
 
+    def lookup_encoder_class(self, column_type):
+        path = 'lightwood.encoders.{type}'.format(type=column_type)
+        module = importlib.import_module(path)
+        if hasattr(module, 'default'):
+            encoder_class = importlib.import_module(path).default
+        else:
+            raise ValueError(
+                'No default encoder for {type}'.format(type=column_type))
+        return encoder_class
```

Comment:
I know this was not something your introduced, but, I  feel like adding the `ValueError` raising here is superfluous, since this edge case is not even one we could encounter at the moment and the default exception thrown should be sufficient for the user to figure out what's happening (and again, I would expect most people to run into this anyway).

If so we could replace this whole block with `return importlib.import_module(path).default`

But your call here, I'm fine with it either way.

Reply:
Agreed. Great, less code